src/main/java/org/disq_bio/disq/impl/file/BaiMerger.java

```diff
@@ -57,33 +62,44 @@ public void mergeParts(
       throws IOException {
     logger.info("Merging .bai files in temp directory {} to {}", tempPartsDirectory, outputFile);
     List<String> parts = fileSystemWrapper.listDirectory(conf, tempPartsDirectory);
-    List<String> baiParts = getBaiParts(parts);
-    if (partLengths.size() - 2 != baiParts.size()) { // don't count header and terminator
+    List<String> filteredParts =
+        parts
+            .stream()
+            .filter(f -> f.endsWith(BAMIndex.BAMIndexSuffix))
+            .collect(Collectors.toList());
+    if (partLengths.size() - 2 != filteredParts.size()) { // don't count header and terminator
       throw new IllegalArgumentException(
           "Cannot merge different number of BAM and BAI files in " + tempPartsDirectory);
     }
-    try (OutputStream out = fileSystemWrapper.create(conf, outputFile)) {
-      List<SeekableStream> baiStreams = new ArrayList<>();
-      for (String baiPart : baiParts) {
-        baiStreams.add(fileSystemWrapper.open(conf, baiPart));
+    int i = 0;
+    ExecutorService executorService = Executors.newFixedThreadPool(8);
```

Comment:
This code is duplicated in `SbiMerger` and `TbiMerger`. A common superclass (`IndexMerger`) would help improve.

Reply:
+1

Reply:
I've now added `IndexMerger` to remove code duplication.